src/main/kotlin/io/github/zap/build/gradle/convention/ProjectExtension.kt

```diff
@@ -169,4 +171,99 @@ fun Project.configureShadow(destination: File,
     tasks.named("build", DefaultTask::class.java) {
         it.dependsOn(sj)
     }
+}
+
+fun Project.qs (name: String = "", options: ExternalModuleDependency.() -> Unit = {}) : ExternalModuleDependency.() -> Unit {
+    return {
+        val actualName = name.ifEmpty { this.name }
+        version {
+            if (isLocalTag(actualName)) {
+                it.require(getLocalVersion(actualName))
+            } else {
+                if(version != null && version!!.isNotEmpty()) {
+                    ext["qs-$actualName-ver"] = version
+                } else {
+                    val ver = getVerFromAncestor(actualName)
+                    if (ver != null) {
+                        it.require(ver)
+                    } else {
+                        logger.log(LogLevel.ERROR, "No version specified for dependency $actualName, fallback to 0.0.0-SNAPSHOT")
+                        it.require("0.0.0-SNAPSHOT")
+                    }
+                }
+            }
+        }
+        options()
+    }
+}
+
+@Suppress("UNCHECKED_CAST")
+internal fun Project.getQsLocals(): Set<String>? {
+    if(ext.has("qs-useLocals")) {
+        val cached = ext["qs-useLocals"]
+        if(cached is Set<*>) {
+            return cached as Set<String>
+        }
+    }
+
+    val useLocalClause = project.findProperty("qs.useLocal") as String?
+    if(useLocalClause == null) {
+        ext["qs-useLocals"] = null
+        return null
+    }
+
+    val useLocals = useLocalClause.split(",")
```

Comment:
could probably use a regex to split without whitespace
also for the sake of things looking nice if you're chaining multiple calls then I think .split should be on new line

Reply:
I usually do newline chaining if it the chains get more than 3 calls

Reply:
there are 4 calls here and usually I see each of them on newline if you are chaining at all

Reply:
![image](https://user-images.githubusercontent.com/33594017/132998604-1c77eb60-14fc-417b-84e2-4602b4ea938e.png)

4 calls?

Reply:
![image](https://user-images.githubusercontent.com/60953955/132998634-f69883d3-c3ff-4bad-b24c-9d08fc585037.png)